packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx

```diff
@@ -164,6 +170,7 @@ export default function HierarchyExplorer({ appId, className }: HierarchyExplore
     }
 
     if (appDom.isPage(node)) {
+      domApi.deselectNode();
```

Comment:
This solves the issue where a node would "stay" even when navigating to a page where that node didn't exist:
<img width="1725" alt="Screenshot 2023-02-14 at 9 02 18 AM" src="https://user-images.githubusercontent.com/463175/218760465-739bb64e-676c-4158-a675-e3329223f0e5.png">

Reply:
Ah yes, interesting 👍 

@apedroferreira Would it make for the future to incorporate the selection state into the `DomView`? We could even think about making selected node deep linkable through a query parameter

Reply:
Yeah, actually it only makes sense for the selected node state to exist inside the main view... So I should change it to be there. I'll create an issue for that and address it.

As for setting things with query parameters, that would be a separate feature but should be totally doable.

Reply:
I have a more global fix for this specific issue in https://github.com/mui/mui-toolpad/pull/1674.